File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    var answer = '';
    
    const hashTable = {};
    
    for(const people of participant){
        if(hashTable[people]){
            hashTable[people]++;
        }else{
            hashTable[people] = 1;
        }

    }
    
    for(const people of completion){
        hashTable[people]--;
    }
    
    
    
    for(const people of participant){
        if(hashTable[people]){
            answer = people;
            break;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dic = {}
    for part in participant:
        dic[part] = dic.get(part, 0) + 1
        
    for name in completion:
        if name in dic:
            dic[name] -= 1
            if(dic[name] == 0):
                del dic[name]
            
    for key in dic:
        return key